File: jft.academia.v01/Engine/jft.academia.v01.Aplicativo/DataViews/RepositorysViews/GruposRepositoryView.cs
using jft.academia.v01.Aplicativo.Common;
using jft.academia.v01.Aplicativo.ModelsEntities;
using jft.academia.v01.Core.Entities;
using jft.academia.v01.Core.EntitiesViews;
using jft.academia.v01.Core.Entities.Interfaces;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

//namespace jft.academia.v01.Aplicativo.DataViews.RepositorysViews
//{
//    public class GruposRepositoryView : 
//        BasePageModelView<Grupos, GruposModel>, 
//        IDataStoreViewAsync<Grupos, GruposModel>
//    {

        



//        public Task<bool> DeleteViewItemAsync(GruposModel item)
//        {
//            throw new NotImplementedException();
//        }

//        public Task<GruposModel> GetEntityToModelAsync(Grupos _entity)
//        {
//            var _r = new GruposModel
//            {
//                id_grupo_atividade = _entity.id_grupo_atividade,
//                nm_grupo_atividade = _entity.nm_grupo_atividade
//            };

//            return Task.FromResult(_r);
//        }

//        public Task<Grupos> GetModelToEntityAsync(GruposModel _entityModel)
//        {
//            throw new NotImplementedException();
//        }

//        public Task<GruposModel> GetViewItemAsync(int id)
//        {
//            var _result = this.dbAcademia.Grupos.GetItemAsync(id).Result;


//            return this.GetEntityToModelAsync(_result);
//        }

//        public Task<List<GruposModel>> GetViewItemsAsync(bool forceRefresh = false)
//        {
             

//            var _result = this.dbAcademia.Grupos.GetItemsAsync().ContinueWith(t =>
//            {
                
//                var lista = t.Result;
//                var listAtividadesView = new List<GruposModel>();
                
//                foreach (var _item in lista)
//                {

//                    var _view = this.GetEntityToModelAsync(_item).Result;
//                    listAtividadesView.Add(_view);

//                }
//                return listAtividadesView;
//            });


//            return _result;


//        }

//        public Task<bool> SaveViewItemAsync(GruposModel item)
//        {
//            throw new NotImplementedException();
//        }


//    }
//}

File: jft.academia.v01/Engine/jft.Academia.v01.Infra01/DataViews/RepositorysViews/GruposRepositoryView.vb
Imports jft.Academia.v01.Core.Entities
Imports jft.Academia.v01.Core.Entities.Interfaces
Imports jft.Academia.v01.Core.EntitiesViews
Imports jft.Academia.v01.Infra01.DataViews.Common

Namespace DataViews



    Namespace RepositorysViews



        Public Class GruposRepositoryView
            Inherits BaseRepositoryView(Of Grupos, GruposModel)
            Implements IDataStoreViewAsync(Of Grupos, GruposModel)





            'Public Function DeleteViewItemAsync(item As GruposModel) As Task(Of Boolean) Implements IDataStoreViewAsync(Of Grupos, GruposModel).DeleteViewItemAsync
            '    Throw New NotImplementedException()
            'End Function

            Public Overloads Function GetEntityToModelAsync(_entity As Grupos) As Task(Of GruposModel) Implements IDataStoreViewAsync(Of Grupos, GruposModel).GetEntityToModelAsync
                Dim _r = New GruposModel With {
                    .id_grupo_atividade = _entity.id_grupo_atividade,
                    .nm_grupo_atividade = _entity.nm_grupo_atividade
                }

                Return Task.FromResult(_r)
            End Function

            'Public Function GetModelToEntityAsync(_entityModel As GruposModel) As Task(Of Grupos) Implements IDataStoreViewAsync(Of Grupos, GruposModel).GetModelToEntityAsync
            '    Throw New NotImplementedException()
            'End Function

            Public Overloads Function GetViewItemAsync(id As Integer) As Task(Of GruposModel) Implements IDataStoreViewAsync(Of Grupos, GruposModel).GetViewItemAsync
                Dim _result = Me.dbAcademia.Grupos.GetItemAsync(id).Result


                Return Me.GetEntityToModelAsync(_result)
            End Function

            Public Overloads Function GetViewItemsAsync(Optional forceRefresh As Boolean = False) As Task(Of List(Of GruposModel)) Implements IDataStoreViewAsync(Of Grupos, GruposModel).GetViewItemsAsync


                Dim _result = Me.dbAcademia.Grupos _
                    .GetItemsAsync().ContinueWith(Function(t)

                                                      Dim lista = t.Result
                                                      Dim listAtividadesView = New List(Of GruposModel)()

                                                      For Each _item In lista

                                                          Dim _view = Me.GetEntityToModelAsync(CType(_item, Grupos)).Result
                                                          listAtividadesView.Add(_view)

                                                      Next
                                                      Return listAtividadesView
                                                  End Function)


                Return _result


            End Function

            'Public Function SaveViewItemAsync(item As GruposModel) As Task(Of Boolean) Implements IDataStoreViewAsync(Of Grupos, GruposModel).SaveViewItemAsync
            '    Throw New NotImplementedException()
            'End Function

        End Class


    End Namespace

End Namespace